devnotes/tpm.md

```diff
@@ -0,0 +1,228 @@
+| Needs OS driver | Inverted | Producing OS driver is necessary                         |
+| Fully open      | Yes      | Whether involved hardware and software are fully open    |
+
+## Approaches
+
+### I2C TPM module
+
+Hardware module with an uncommon interface (TPMs for PC mostly use LPC or SPI).
+Known chips of this kind are/were produced by Nuvoton, ST, Infineon,
+Atmel (AT97SC3205T).
+
+* **Pros**
+  * Talos II has I2C header exposed on the board.
+  * Best option if can be bought.
+
+* **Cons**
+  * No datasheets for Nuvoton chips.
+  * Limited or no availability of ready to use modules.
+  * Might require design and production of new TPM module with I2C interface.
+
+* **Risks**
+  * Hard to find and this might not improve in the future.
+
+* **Implementation effort**
+  * See [TPM over I2C](tpm_over_i2c.md).
+
+* **Other notes**
+  * ST33 has Linux driver for TPM 2.0
+  * Bought modules won't have OSS firmware.
+
+******
+### Software TPM on BMC
+
+TPM functionality exposed to the POWER9 CPU via LPC gadget device from BMC.
+As secure as BMC is.
+
+* **Pros**
+  * No additional hardware needed.
+  * Open implementation.
+
+* **Cons**
+  * Need to secure the BMC and manage that properly.
+
+* **Risks**
+  * BMC might not have been designed to fulfill security requirements.
+
+* **Implementation effort**
+  * Implement fTPM in BMC.
+  * LPC gadget BMC -> POWER.
+  * No existing driver. Maybe need to hack up existing LPC driver.
+
+* **Other notes**
+  * We could go full RO with coreboot if desired (by sacrificing some
+    performance for MVPD).
+
+******
+### FPGA translator (LPC TPM cycle - LPC TPM cycle)
+
+New piece of hardware that addresses LPC cycles issues.
+
+* **Pros**
+  * Conventional TPM modules.
+  * No need for a new driver.
+
+* **Cons**
+  * Need to design and manufacture this hardware.
+
+* **Risks**
+
+* **Implementation effort**
+  * Slightly easier than the fTPM on BMC - but not much.
+
+* **Other notes**
+
+******
+### FPGA translator (SPI - LPC TPM cycle)
+
+New piece of hardware that addresses LPC cycles issues.
+
+* **Pros**
+  * Conventional TPM modules.
+  * No need for a new driver.
+
+* **Cons**
+  * Need to design and manufacture this hardware.
+
+* **Risks**
+
+* **Implementation effort**
+  * Slightly easier than the fTPM on BMC - but not much.
+
+* **Other notes**
+
+******
+### Open TPM on MCU (e.g. [lpnTPM])
+
+Find an existing open hardware TPM and possibly extend it for I2C interface.
+
+* **Pros**
+  * Open hardware.
+
+* **Cons**
+  * Need to implement and manufacture TPM module.
+
+* **Risks**
+  * I2C is not in the scope as of now.
+
+* **Implementation effort**
+  * Slightly easier than the fTPM on BMC - but not much.
+
+* **Other notes**
+
+******
+### TPM on [LibreBMC][LibreBMC-site]
+
+Embed TPM into an alternative BMC hardware that's still in development?
+The BMC itself is implemented as a custom extension card defined by DS-CMS?
```

Comment:
I think it can't be plugged in at all... See https://www.opencompute.org/documents/ocp-dc-scm-spec-rev-1-0-pdf, "3.4.12 Standby Power and Boot Sequence" for booting and the rest for card specification. It needs compatible motherboards.

Reply:
@krystian-hebel are we sure? https://www.talospace.com/2021/10/first-flight-of-kestrel-fpga-openpower.html it has access to power regulator

Reply:
@pietrushnic the link you provided doesn't mention LibreBMC.

Reply:
@krystian-hebel are we looking at the same thing?
![librebmc_kestrel](https://user-images.githubusercontent.com/1071894/165400085-6f381f94-e68a-4f79-9ad0-f057f584c83a.png)

Reply:
I meant this coincidence: https://www.talospace.com/2021/05/librebmc-announced-end-of-kestrel.html

Reply:
From what I understand LibreBMC can't be used with already existing boards. Kestrel on the other hand can be used with Talos II as is, but I think it doesn't have TPM either and ASpeed BMC needs to be disabled to not interfere, which complicates things. Plus coreboot port might behave differently with Kestrel.

Reply:
@SergiiDmytruk please add that to documentation. Maybe economical feasibility of these paths is not the best, maybe there are technical obstacles, but good documentation/analysis should say that.

To close this item we should confirm or disprove @krystian-hebel challenge about powering up. I tried, but I'm definitely not the expert here.

Reply:
Yes, it can power on the host. Referenced section says that this card should have 12V power before the host is powered:

> The DC-SCI interface supports 12V for powering the DC-SCM. Sequencing is specified such that STBY
power on the DC-SCM is enabled and valid before STBY power on the HPM.